File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        
        if(s.length()!=t.length()){
            return false;
        }
        unordered_map<char,int> mp;
        
        for(int i=0;i<s.length();i++){
           mp[s[i]]++;
           mp[t[i]]--;
        }
        
        for(int i=0;i<mp.size();i++){
            if(mp[s[i]]!=0)
                return false;
        }
        return true;
        
    }
};

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        
        if(s.length()!=t.length()){
            return false;
        }
        
        HashMap<Character,Integer> mp = new HashMap<>();
        
        for(int i=0;i<s.length();i++){
            mp.put(s.charAt(i),mp.getOrDefault(s.charAt(i),0)+1);
            mp.put(t.charAt(i),mp.getOrDefault(t.charAt(i),0)-1);
        }
        
        for(Character key:mp.keySet()){
            if(mp.get(key)!=0){
                return false;
            }
        }
        
        return true;
        
        
    }
}